### ISSUE TITLE
Autocomplete and search results inconsistent

### ISSUE THREAD

**Issue by wecreatedigital** [Smile-SA/elasticsuite#2169]
Hi there, thank you for ElasticSuite, it is a great addition to any Magento 2 website!

I have a problem understanding why the display of results between the autocomplete and the results on a search index page are different. This can be demonstrated with this recording: https://gyazo.com/70c9050a0736c3915df7ca4937ad6b7a - you'll see the order of products is very different but surely they would be the same?

When I completely disable ElasticSuite and the issue does not occur.

### Preconditions

Magento Version : CE 2.4.2

ElasticSuite Version : 2.10.4

Environment : Both developer and production

Third party modules : 

Amasty_Base
Amasty_Conditions
Amasty_CommonRules
Amasty_Faq
Amasty_InvisibleCaptcha
Amasty_ShippingArea
Amasty_ShippingTableRates
Amasty_Shiprestriction
Ebizmarts_MailChimp
FishPig_WordPress
FishPig_WordPress_ACF
FishPig_WordPress_Multisite
FishPig_WordPress_PluginShortcodeWidget
FishPig_WordPress_PostTypeTaxonomy
FishPig_WordPress_Yoast
Loqate_Tag
MageSuite_Magepack
Magebees_Navigationmenu
Magefan_Community
Magefan_WysiwygAdvanced
Mageplaza_Core
Mageplaza_GoogleRecaptcha
Mageplaza_Seo
Mageplaza_Sitemap
Mageplaza_Smtp
RedChamps_Core
RedChamps_CurrentProductRegistry
RedChamps_SimilarProducts
RedChamps_DiscontinuedProducts
Smile_ElasticsuiteAdminNotification
Smile_ElasticsuiteCore
Smile_ElasticsuiteCatalog
Smile_ElasticsuiteCatalogRule
Smile_ElasticsuiteCatalogOptimizer
Smile_ElasticsuiteTracker
Smile_ElasticsuiteThesaurus
Smile_ElasticsuiteSwatches
Smile_ElasticsuiteIndices
Smile_ElasticsuiteAnalytics
StripeIntegration_Payments
Swissup_Ajaxlayerednavigation
Swissup_Core
Swissup_Checkout
Swissup_CheckoutRegistration
Swissup_Codemirror
Swissup_Firecheckout
Swissup_FirecheckoutIntegrations
Swissup_Image
Swissup_Marketplace
Swissup_Pagespeed
Swissup_Rtl
Swissup_SlickCarousel
Swissup_Stickyfill
Swissup_SubscribeAtCheckout
Swissup_Swiper
Swissup_Taxvat
Swissup_Tippyjs
Unicode_Core
Unicode_ProductAttachment
Wyomind_Core
Wyomind_SimpleGoogleShopping
Wyomind_Watchlog

### Steps to reproduce

1. Start typing phrase such as '660-fc' as per the above recording
2. Acknowledge the order of products in autocomplete
3. Hit submit to make the search
4. Review the order of products on result page vs the order in the autocomplete

### Expected result

The order of the autocomplete should be the same as initial search results page.

### Actual result

The order of the autocomplete vs the search results page was different.

**Comment by romainruaud** [comment]
TL;DR version : this is working as intended, and it's done like this by design.

First type, you type "660-fc", this will be **extended to all existing search terms beginning with this string** : "660-fc1" and "660-fc2".

=> In autocompletion, **we widen the search, because we consider the user is typing and might not have finished writing** => we widen to all search terms beginning with the same string as the one currently being typed. So "pan" will find "pans", "pants", "panda", "pancakes", and so on... exactly as per your example.

=> But on the search result page, we assume **the user has finished typing** (because he did hit ENTER and submit the form). So in this case, we favor exact matches to remain the most relevant. In your case, we search only for "660-fc".
